### ISSUE TITLE
Improve knative broker integration

### ISSUE THREAD

**Issue by nicolaferraro** [apache/camel-k-runtime#326]
When doing something like:

```
from("knative:event/x")
  .transform()
  .to("knative:event/y")
```

Then the event is delivered to the same service again. The reason is that the header containing the event type (x) overrides the event type "y" provided in the producer endpoint.

It's not so clear what happens.. Maybe we should make such override not enabled by default.

**Comment by davsclaus** [comment]
Yeah in other components we have a special header named CamelOverrideXXX etc to indicate this fact

**Comment by lburgazzoli** [comment]
I gues in this case we can safely ignore such header as the event is already specified in the URI ? in case you want this dynamic I guess we can use a simple expression and a `tod` ?

**Comment by nicolaferraro** [comment]
I'm not sure `toD` works in this case because we have a strict relationship between endpoints and services in the KNATIVE_CONFIG env and the dynamic endpoint is not passed by the operator.

But in the case of the broker, the event name should not be mandatory to lookup the instance.
E.g. I might define a producer like `.to("knative:event")` and use the header to determine the event type, but this is not possible (null pointer exception).

**Comment by lburgazzoli** [comment]
so if the event is set on the endpoint, then use it otherwise use the header one should work
